mobile/src/screens/BookScreen.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { useState, useContext, useEffect } from 'react';
+import { Text, View, StyleSheet, Image, ScrollView } from 'react-native';
+import { useRoute } from '@react-navigation/native';
+import { MarkdownView } from 'react-native-markdown-view';
+import YoutubePlayer from 'react-native-youtube-iframe';
+import { BookScreenRouteProps } from '../navigation/HomeStackNavigator';
+import { LoadingCircle } from '../components/LoadingCircle';
+import { APIContext } from '../context/APIContext';
+import { I18nContext } from '../context/I18nContext';
+import { TextStyles } from '../styles/TextStyles';
+import { ButtonGroup } from '../components/ButtonGroup';
+import { LanguageButtons } from '../components/LanguageButtons';
+
+/**
+ * Individual book view displaying book details
+ */
+export const BookScreen: React.FC = () => {
+  // get book id and langs from route params
+  const route = useRoute<BookScreenRouteProps>();
+  const { id, langs } = route.params;
+
+  const client = useContext(APIContext);
+  const i18nCtx = useContext(I18nContext);
+
+  // book screen states
+  const [loading, setLoading] = useState(true);
+  const [language, setLanguage] = useState(i18nCtx.locale.substring(0, 2));
```

Comment:
So I think the default lang we want here should be (in order of precedence):

1. User's current lang if it exists in book.languages
2. English if it exists in book.languages
3. book.languages[0]

I think you have everything you need to determine this default value before the call to useState, you should be able to do something like:

```ts
const currentLang = i18nCtx.locale.substring(0,2);
const defaultLang = langs.includes(currentLang) ? currentLang : langs.includes('en') ? 'en' : langs[0];
...
const [language, setLanguage] = useState(defaultLang);
```